executorlib/interactive/executor.py

```
@@ -3,110 +1,274 @@
+
+def create_executor(
+    max_workers: int = 1,
+    backend: str = "local",
+    max_cores: int = 1,
+    cache_directory: Optional[str] = None,
+    cores_per_worker: int = 1,
+    threads_per_core: int = 1,
+    gpus_per_worker: int = 0,
+    cwd: Optional[str] = None,
+    openmpi_oversubscribe: bool = False,
+    slurm_cmd_args: list[str] = [],
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)